### PR TITLE
Check in protos and add a make target to import them into target repos

### PR DESCRIPTION
**Pull request by sdake**
Check in protos for api and apimachinery
    
    This adds a new make target make update-common-protos. I am not
    sure if this works yet, as it needs to hit the common repo for proper
    testing.
    
    * Add phony target `make update-common-protos` which adds a protos dir
    * Pin to (master) currently (will determine SHA in a followup)
    * Pull protos for following repos:
       * https://github.com/protocolbuffers/protobuf
       * https://github.com/kubernetes/api.git
       * https://github.com/kubernetes/apimachinery.git
       * https://github.com/protobuf/gogo.git
       * https://github.com/envoyproxy/protoc-gen-validate/

### CODE REVIEW

File: get_protos.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+# Retrieve a copy of Googles's protobufs
+pushd "${TEMPDIR_GOOGLE}" || exit
+git clone --depth 1 --single-branch --branch "${GOOGLE_TAG}" https://github.com/protocolbuffers/protobuf.git
+pushd protobuf/src/google || exit
+find . -name \*proto | cpio -pdm "${REPODIR}"/protos/google
+popd || exit
+popd || exit
+
+# Retrieve a copy of Googles's protobufs
+pushd "${TEMPDIR_GOOGLEGRPC}" || exit
```
[the remark this answers is not in the document]
Comment (sdake) [reply get_protos.sh]: fixed.

File: files/proto/get_protos.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+mkdir -p usr/include/protobuf/k8s.io/api
+pushd  ${TEMPDIRAPI}
+git clone --depth 1 --single-branch --branch master https://github.com/kubernetes/api.git
+popd
+find ${TEMPDIRAPI} -name \*proto -exec cp -a --parents {} usr/include/protobuf/k8s.io/api \;
+
+mkdir -p usr/include/protobuf/k8s.io/apimachinery
+pushd  ${TEMPDIRAPIMACHINERY}
+git clone --depth 1 --single-branch --branch master https://github.com/kubernetes/api.git
```
[the remark this answers is not in the document]
Comment (sdake) [reply files/proto/get_protos.sh]: pinning capability added.

File: files/common/Makefile.common.mk [hunk 3]
```diff
@@ -57,4 +57,11 @@ update-common:
 	@cp -r common-files/files/* .
 	@rm -fr common-files
 
-.PHONY: lint-dockerfiles lint-scripts lint-yaml lint-copyright-banner lint-go lint-pyhton lint-helm format-go format-python update-common
+update-common-protos:
+	@git clone --depth 1 --single-branch --branch master https://github.com/istio/common-files
+	@cd common-files ; git rev-parse HEAD >files/common/.commonfiles.sha
```
[the remark this answers is not in the document]
Comment (sdake) [reply files/common/Makefile.common.mk]: fixed.

File: get_protos.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+REPODIR=$(pwd)
+
+TEMPDIR_API=$(mktemp -d /tmp/api-XXXXXXXX)
+TEMPDIR_APIMACHINERY=$(mktemp -d /tmp/apimachinery-XXXXXXXX)
+TEMPDIR_GOGO=$(mktemp -d /tmp/gogo-XXXXXXXX)
+
+API_TAG=master
```
[the remark this answers is not in the document]
Comment (sdake) [reply get_protos.sh]: ok - pinned :)

**Comment by geeknoid** [comment]
Here's the most recent build-tools image: gcr.io/istio-testing/build-tools:2019-08-29T13-57-48

**Pull request merged**